app/src/middleware/authorization.js

```diff
@@ -44,6 +52,56 @@ const _checkPermission = async ({ currentObject, currentUser, params }, permissi
   log.debug('Missing user identification', { function: '_checkPermission' });
   return result;
 };
+/**
+ * @function checkS3BasicAccess
+ * Checks and authorized access to perform operation for s3 basic authentication request
+ * @param {object} req Express request object
+ * @param {object} _res Express response object
+ * @param {function} next The next callback function
+ * @returns {function} Express middleware function
+ * @throws The error encountered upon failure
+ * @returns
+ */
+const checkS3BasicAccess = async (req, _res, next) => {
+  const authType = req.currentUser ? req.currentUser.authType : undefined;
+  const bucketSettings = req.currentUser?.bucketSettings ? req.currentUser.bucketSettings : undefined;
+
+  if (getConfigBoolean('basicAuth.s3AccessMode') && authType === AuthType.BASIC && bucketSettings) {
+    // determine which buckets relate to the request
+    let bucketIds = mixedQueryToArray(req.query.bucketId) || mixedQueryToArray(req.params.bucketId) || req.body.bucketId;
+    const objIds = mixedQueryToArray(req.query.objectId) || mixedQueryToArray(req.params.objectId) || req.body.objectId;
+    const versionIds = mixedQueryToArray(req.query.versionId);
+    const s3VersionIds = mixedQueryToArray(req.query.s3VersionId);
+
+    if (!bucketIds?.length) {
+      if (objIds?.length || versionIds?.length || s3VersionIds?.length) {
+        const objects = await objectService.searchObjects({
+          id: objIds, versionId: versionIds, s3VersionId: s3VersionIds
+        });
+        bucketIds = objects.data.map(i => i.bucketId);
+      }
+    }
+
+    // filter request by buckets matching provided credentials
+    try {
+      const bucketData = {
+        bucketId: bucketIds,
+        bucket: bucketSettings.bucket,
+        endpoint: bucketSettings.endpoint,
```

Comment:
If the endpoint as stored in the DB does not have a trailing slash but `x-amz-endpoint` does, I can retrieve *all* COMS buckets across all users.

e.g. `bucket.endpoint` (in the database) is `http://example.com`, but in the API call I have the header `x-amz-endpoint: http://example.com/`.

`utils.js` has a `stripDelimit()` helper function that should fix things:
```js
        endpoint: stripDelimit(bucketSettings.endpoint)
```